src/app/shared/services/task/task.service.ts

```diff
@@ -1,26 +1,117 @@
+  appFields: IAppConfig["APP_FIELDS"];
+  taskGroups: any[];
+  taskGroupsHashmap: Record<string, any>;
+  constructor(
+    private templateFieldService: TemplateFieldService,
+    private appConfigService: AppConfigService,
+    private appDataService: AppDataService
+  ) {
+    this.subscribeToAppConfigChanges();
+  }
+
+  async init() {
+    await this.getListOfTaskGroups();
+    this.evaluateHighlightedTaskGroup();
+  }
+
+  /** Get the list of highlight-able task groups, from the relevant data_list */
+  private async getListOfTaskGroups() {
+    // TODO: This should be set from the template/skin level
+    const taskGroupsListName = "workshop_tasks";
+    const taskGroupsDataList = await this.appDataService.getSheet("data_list", taskGroupsListName);
+    this.taskGroups = taskGroupsDataList.rows;
+    this.taskGroupsHashmap = arrayToHashmap(this.taskGroups, "id");
+  }
+
+  /** The _highlighted_task_group should always be the ID of the highest
+   * priority task_group that is not completed and not skipped
+   * NB "highest priority" is defined as having the lowest numerical value for the "number" column
+   */
+  public async evaluateHighlightedTaskGroup() {
+    const taskGroupsNotCompletedAndNotSkipped = clone(this.taskGroups).filter((taskGroup) => {
+      return (
+        !this.templateFieldService.getField(taskGroup.completed_field) &&
+        !this.templateFieldService.getField(taskGroup.skipped_field)
+      );
+    });
+    // If all task groups are completed or skipped (e.g. when user completes final task group),
+    // then set highlighted task group to undefined
+    if (taskGroupsNotCompletedAndNotSkipped.length === 0) {
+      this.templateFieldService.setField(this.appFields.HIGHLIGHTED_DATA_ID, "undefined");
+    }
+    const highestPriorityTaskGroup = taskGroupsNotCompletedAndNotSkipped.reduce(
+      (highestPriority, taskGroup) => {
+        return highestPriority.number < taskGroup.number ? highestPriority : taskGroup;
+      }
+    );
+    this.templateFieldService.setField(
+      this.appFields.HIGHLIGHTED_DATA_ID,
+      highestPriorityTaskGroup.id
+    );
+    console.log("[HIGHLIGHTED TASK GROUP] - ", highestPriorityTaskGroup.id);
+  }
+
+  public getHighlightedTaskGroup() {
+    return this.templateFieldService.getField(this.appFields.HIGHLIGHTED_DATA_ID);
+  }
 
-  public getHighlightedTask() {
-    return this.templateFieldService.getField("highlighted_task");
+  /** For a given task groups list, @return the index of the highlighted task within that list */
+  public async getHighlightedTaskGroupIndex(taskGroupsList: string) {
+    const taskGroupsDataList = await this.appDataService.getSheet("data_list", taskGroupsList);
+    const arrayOfIds = taskGroupsDataList.rows.map((taskGroup) => taskGroup.id);
+    console.log("arrayOfIds:", arrayOfIds);
+    console.log("HighlightedTaskGroup:", this.getHighlightedTaskGroup());
+    return arrayOfIds.indexOf(this.getHighlightedTaskGroup());
```

Comment:
nit(non-blocking)
Youd should make the call to getHighlightedTaskGroup at the start. That way if it is "undefined" (or assuming you update as suggested, "") you can return a null value immediately and in the carousel.component use the null value instead of checking for -1